sympy/matrices/expressions/tests/test_derivatives.py

```diff
@@ -137,26 +144,34 @@ def test_matrix_derivative_vectors_and_scalars():
     # Cookbook example 83:
     expr = (X*b + c).T*D*(X*b + c)
     assert expr.diff(X) == D*(X*b + c)*b.T + D.T*(X*b + c)*b.T
+    assert expr[0, 0].diff(X[m, n]).doit()
```

Comment:
Is there something missing from this assert?

Reply:
Yes, there is a problem with the simplification of the Kronecker delta... the expression is not optimal. I will try to fix this.